src/index.ts

```diff
@@ -285,6 +285,10 @@ const plugin: JupyterFrontEndPlugin<void> = {
         }
 
         await switchKernel(panel, kernel);
+
+        const url = new URL(window.location.href);
+        url.searchParams.set('kernel', kernel);
+        window.history.replaceState({}, '', url.toString());
```

Comment:
Do we need this in the URL? I am worried that this will complicate things without much benefit?